src/OSPSuite.Core/Domain/Services/KeywordReplacerTask.cs

```diff
@@ -271,11 +279,13 @@ private void replaceInUsingFormula(IUsingFormula usingFormula, IContainer rootCo
 
       public void ReplaceIn(IParameter parameter, IContainer rootContainer)
       {
-         var moleculeAmount = parameter.ParentContainer as IMoleculeAmount;
-         if (moleculeAmount == null)
-            replaceIn(parameter, rootContainer, string.Empty);
-         else
-            ReplaceIn(parameter, rootContainer, moleculeAmount.Name);
+         var parameterContainer = parameter.ParentContainer;
+
+         //Global molecule container or local molecule amount container
+         var isInMolecule = parameterContainer.IsAnImplementationOf<IMoleculeAmount>() ||
```

Comment:
@Yuri05 This is the issue here being fixed. The rest is noise.
In short, global molecule parameters referencing other global molecule parameters using keyword would not work. Replacement was not being done for MOLECULE in that case